subsys/net/lib/sockets/socketpair.c

```diff
@@ -0,0 +1,1132 @@
+				default: {
+					__ASSERT(false,
+						"unrecognized result: %d",
+						result);
+					continue;
+				}
+			}
+
+			/* SPAIR_SIG_DATA was received */
+			break;
+		}
+	}
+
+	res = k_pipe_get(&spair->recv_q, (void *)buffer, count, &bytes_read,
+			 1, K_NO_WAIT);
+	__ASSERT(res == 0, "k_pipe_get() failed: %d", res);
+
+	if (is_connected) {
+		res = k_poll_signal_raise(&spair->read_signal, SPAIR_SIG_DATA);
+		__ASSERT(res == 0, "k_poll_signal_raise() failed: %d", res);
+	}
+
+	res = bytes_read;
+
+out:
+
+	if (spair != NULL && have_local_sem) {
+		k_sem_give(&spair->sem);
+	}
+
+	return res;
+}
+
+static int zsock_poll_prepare_ctx(struct spair *const spair,
+				  struct zsock_pollfd *const pfd,
+				  struct k_poll_event **pev,
+				  struct k_poll_event *pev_end)
+{
+	int res;
+
+	struct spair *remote = NULL;
+	bool have_remote_sem = false;
+
+	if (pfd->events & ZSOCK_POLLIN) {
+
+		/* Tell poll() to short-circuit wait */
+		if (sock_is_eof(spair)) {
+			res = -EALREADY;
+			goto out;
+		}
+
+		if (*pev == pev_end) {
+			res = -ENOMEM;
+			goto out;
+		}
+
+		/* Wait until data has been written to the local end */
+		(*pev)->obj = &spair->write_signal;
+	}
+
+	if (pfd->events & ZSOCK_POLLOUT) {
+
+		/* Tell poll() to short-circuit wait */
+		if (!sock_is_connected(spair)) {
+			res = -EALREADY;
+			goto out;
+		}
+
+		if (*pev == pev_end) {
+			res = -ENOMEM;
+			goto out;
+		}
+
+		remote = z_get_fd_obj(spair->remote,
+			(const struct fd_op_vtable *)
+			&spair_fd_op_vtable, 0);
+
+		__ASSERT(remote != NULL, "remote is NULL");
+
+		res = k_sem_take(&remote->sem, K_FOREVER);
+		if (res < 0) {
+			goto out;
+		}
+
+		have_remote_sem = true;
+
+		/* Wait until data has been read from the remote end */
+		(*pev)->obj = &remote->read_signal;
+	}
+
+	(*pev)->type = K_POLL_TYPE_SIGNAL;
+	(*pev)->mode = K_POLL_MODE_NOTIFY_ONLY;
+	(*pev)->state = K_POLL_STATE_NOT_READY;
+	k_poll_signal_reset((*pev)->obj);
+
+	(*pev)++;
+
+	res = 0;
+
+out:
+
+	if (remote != NULL && have_remote_sem) {
+		k_sem_give(&remote->sem);
+	}
+
+	return res;
+}
+
+static int zsock_poll_update_ctx(struct spair *const spair,
+				 struct zsock_pollfd *const pfd,
+				 struct k_poll_event **pev)
+{
+	int res;
+
+	int signaled;
+	int result;
+
+	struct spair *remote = NULL;
+	bool have_remote_sem = false;
+
+	if (pfd->events & ZSOCK_POLLOUT) {
+		if (!sock_is_connected(spair)) {
+			pfd->revents |= ZSOCK_POLLHUP;
+			goto check_pollin;
+		}
+
+		remote = z_get_fd_obj(spair->remote,
+			(const struct fd_op_vtable *)
+			&spair_fd_op_vtable, 0);
+
+		__ASSERT(remote != NULL, "remote is NULL");
+
+		res = k_sem_take(&remote->sem, K_FOREVER);
+		if (res < 0) {
+			/* if other end is deleted, this might occur */
+			goto pollout_done;
+		}
+
+		have_remote_sem = true;
+
+		if (spair_write_avail(spair) > 0) {
+			pfd->revents |= ZSOCK_POLLOUT;
+			goto pollout_done;
+		}
+
+		/* check to see if op was canceled */
+		signaled = false;
+		k_poll_signal_check(
+			&remote->read_signal,
```

Comment:
wouldn't this line fit in prevous one as line 812 looks quite short

Reply:
Will check. But generally, until #21392 and #23823 are done, I just do what chrckpatch tells me to do.